app/controllers/ruleset_controller.rb

```diff
@@ -30,38 +30,27 @@ def create
       return render @ruleset_name
     end
 
-    # Make the API calls in a begin/rescue block so that
-    # we don't crash out if the API or CLI calls fail.
-    begin
-      if @ruleset_name == 'security_ruleset' # Call 42Crunch service
-        if ENV['BYPASS_API'] == 'true'
-          crunch_results = File.read('fakecrunchresults.json')
-        else
-          crunch_results = Linters::CrunchApi.new.lint_to_json(ruleset_params[:oas_file].tempfile)
-        end
-        crunch_json = JSON.parse(crunch_results)
-        @score = crunch_json['score']
-        @openapi = crunch_json['openapiState']
-        @counter = crunch_json['issueCounter']
-        @criticality = crunch_json['criticality']
-        @issues = Array.new
-        crunch_json['data']['issues'].each do |issue|
-          lines = Array.new
-          issue[1]['issues'].each do | occurrence|
-            lines.push(occurrence['pointer'])
-          end
-          @issues.push([issue[1]['criticality'],issue[1]['description'], lines])
+    if @ruleset_name == 'security_ruleset' # Call 42Crunch service
+      if ENV['BYPASS_API'] == 'true'
+        crunch_results = File.read('fakecrunchresults.json')
+      else
+        crunch_results = Linters::CrunchApi.new(file: ruleset_params[:oas_file]).lint_to_json
+      end
+      crunch_json = JSON.parse(crunch_results)
+      @score = crunch_json['score']
+      @openapi = crunch_json['openapiState']
+      @counter = crunch_json['issueCounter']
+      @criticality = crunch_json['criticality']
+      @issues = Array.new
+      crunch_json['data']['issues'].each do |issue|
```

Comment:
Rather than declaring an array you can use something like `each_with_object`:

`crunch_json['data']['issues'].each_with_object([]) do |issue, issues|...`

Reply:
I'd suggest having a look at the solution I suggested here: https://github.com/co-cddo/api-spec-linter/pull/14#discussion_r1099979031

`each_with_object` is a handy operator, but as `crunch_json['data']['issues']` is already an array, `map` is a little simpler.

Reply:
Also part of my refactor but not for this PR - I did it this way originally because I was just "thinking out loud" in terms of working out which bits of JSON I needed and I found it quick and easy to just refer to keys... ultimately we probably need to agree a common schema for the view that we use for both spectral and 42crunch.